x509/x509.go

```diff
@@ -1445,6 +1445,49 @@ func ParseTBSCertificate(asn1Data []byte) (*Certificate, error) {
 		TBSCertificate: tbsCert})
 }
 
+// RebuildRawTBSCertificate regenerates the RawTBSCertificate field of a Certificate based
+// on the current contents of the certificate.  This function does not regenerate the
+// signature value for the certificate.
+func (cert *Certificate) RebuildRawTBSCertificate() error {
+	_, signatureAlgorithm, err := signingParamsForPublicKey(cert.PublicKey, cert.SignatureAlgorithm)
+	if err != nil {
+		return fmt.Errorf("failed to determine signing algorithm: %v", err)
+	}
+
+	asn1Subject, err := asn1.Marshal(cert.Subject.ToRDNSequence())
+	if err != nil {
+		return fmt.Errorf("failed to marshal cert.Subject: %v", err)
+	}
+	asn1Issuer, err := asn1.Marshal(cert.Issuer.ToRDNSequence())
+	if err != nil {
+		return fmt.Errorf("failed to marshal cert.Issuer: %v", err)
+	}
+	cert.ExtraExtensions = cert.Extensions
+	extensions, err := buildExtensions(cert)
+	if err != nil {
+		return fmt.Errorf("failed to rebuild extensions: %v", err)
+	}
+
+	tbs := tbsCertificate{
+		Version:            cert.Version - 1,
+		SerialNumber:       cert.SerialNumber,
+		SignatureAlgorithm: signatureAlgorithm,
+		Issuer:             asn1.RawValue{FullBytes: asn1Issuer},
+		Validity:           validity{cert.NotBefore.UTC(), cert.NotAfter.UTC()},
+		Subject:            asn1.RawValue{FullBytes: asn1Subject},
+		PublicKey:          publicKeyInfo{Raw: cert.RawSubjectPublicKeyInfo},
```

Comment:
Good spot, but this code is gone now.

(The base library `x509.CreateCertificate` misses these fields out, and I adapted that code.)